pytest.deprecated_call doesn't catch PendingDeprecationWarnings

In 2.7.3 it did, but now it doesn't.

```
(tempenv-079b216902410) ~/.v/tempenv-079b216902410 $ pip install pytest==2.7.3
Collecting pytest==2.7.3
  Using cached pytest-2.7.3-py2.py3-none-any.whl
Collecting py>=1.4.29 (from pytest==2.7.3)
  Using cached py-1.4.30-py2.py3-none-any.whl
Installing collected packages: py, pytest
Successfully installed py-1.4.30 pytest-2.7.3
(tempenv-079b216902410) ~/.v/tempenv-079b216902410 $ py.test t.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.10 -- py-1.4.30 -- pytest-2.7.3
rootdir: /Users/vacogaynoa/.virtualenvs/tempenv-079b216902410, inifile:
collected 1 items

t.py .

==================================================== 1 passed in 0.00 seconds ====================================================
(tempenv-079b216902410) ~/.v/tempenv-079b216902410 $ pip install -U pytest
Collecting pytest
  Using cached pytest-2.8.0-py2.py3-none-any.whl
Requirement already up-to-date: py>=1.4.29 in ./lib/python2.7/site-packages (from pytest)
Installing collected packages: pytest
  Found existing installation: pytest 2.7.3
    Uninstalling pytest-2.7.3:
      Successfully uninstalled pytest-2.7.3
Successfully installed pytest-2.8.0
(tempenv-079b216902410) ~/.v/tempenv-079b216902410 $ py.test t.py
====================================================== test session starts =======================================================
platform darwin -- Python 2.7.10, pytest-2.8.0, py-1.4.30, pluggy-0.3.1
rootdir: /Users/vacogaynoa/.virtualenvs/tempenv-079b216902410, inifile:
collected 1 items

t.py F

============================================================ FAILURES ============================================================
_____________________________________________________________ test_f _____________________________________________________________

    def test_f():
>       pytest.deprecated_call(f)
E       AssertionError: <function f at 0x106ce7410> did not produce DeprecationWarning

t.py:11: AssertionError
------------------------------------------------------ Captured stderr call ------------------------------------------------------
/Users/vacogaynoa/.virtualenvs/tempenv-079b216902410/t.py:7: PendingDeprecationWarning: msg
  warnings.warn('msg', PendingDeprecationWarning)
==================================================== 1 failed in 0.00 seconds ====================================================
(tempenv-079b216902410) ~/.v/tempenv-079b216902410 $ cat t.py
import warnings

import pytest


def f():
    warnings.warn('msg', PendingDeprecationWarning)


def test_f():
    pytest.deprecated_call(f)
```

--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,6 +1,11 @@
 2.8.1.dev
 ---------
 
+- 'deprecated_call' is now only satisfied with a DeprecationWarning or
+  PendingDeprecationWarning. Before 2.8.0, it accepted any warning, and 2.8.0
+  made it accept only DeprecationWarning (but not PendingDeprecationWarning).
+  Thanks Alex Gaynor for the issue and Eric Hunsberger for the PR.
+
 - fix issue #1073: avoid calling __getattr__ on potential plugin objects.
   This fixes an incompatibility with pytest-django.  Thanks Andreas Pelme,
   Bruno Oliveira and Ronny Pfannschmidt for contributing and Holger Krekel

--- a/_pytest/recwarn.py
+++ b/_pytest/recwarn.py
@@ -36,7 +36,8 @@ def deprecated_call(func, *args, **kwargs):
         warnings.simplefilter('always')  # ensure all warnings are triggered
         ret = func(*args, **kwargs)
 
-    if not any(r.category is DeprecationWarning for r in wrec):
+    depwarnings = (DeprecationWarning, PendingDeprecationWarning)
+    if not any(r.category in depwarnings for r in wrec):
         __tracebackhide__ = True
         raise AssertionError("%r did not produce DeprecationWarning" % (func,))
 

--- a/testing/test_recwarn.py
+++ b/testing/test_recwarn.py
@@ -115,6 +115,14 @@ class TestDeprecatedCall(object):
         f = lambda: py.std.warnings.warn(PendingDeprecationWarning("hi"))
         pytest.deprecated_call(f)
 
+    def test_deprecated_call_specificity(self):
+        other_warnings = [Warning, UserWarning, SyntaxWarning, RuntimeWarning,
+                          FutureWarning, ImportWarning, UnicodeWarning]
+        for warning in other_warnings:
+            f = lambda: py.std.warnings.warn(warning("hi"))
+            with pytest.raises(AssertionError):
+                pytest.deprecated_call(f)
+
 
 class TestWarns(object):
     def test_strings(self):

--- a/testing/test_recwarn.py
+++ b/testing/test_recwarn.py
@@ -81,7 +81,7 @@ def dep_explicit(i):
 class TestDeprecatedCall(object):
     def test_deprecated_call_raises(self):
         excinfo = pytest.raises(AssertionError,
-                       "pytest.deprecated_call(dep, 3)")
+                                "pytest.deprecated_call(dep, 3)")
         assert str(excinfo).find("did not produce") != -1
 
     def test_deprecated_call(self):
@@ -105,11 +105,15 @@ class TestDeprecatedCall(object):
 
     def test_deprecated_explicit_call_raises(self):
         pytest.raises(AssertionError,
-                       "pytest.deprecated_call(dep_explicit, 3)")
+                      "pytest.deprecated_call(dep_explicit, 3)")
 
     def test_deprecated_explicit_call(self):
         pytest.deprecated_call(dep_explicit, 0)
         pytest.deprecated_call(dep_explicit, 0)
+
+    def test_deprecated_call_pending(self):
+        f = lambda: py.std.warnings.warn(PendingDeprecationWarning("hi"))
+        pytest.deprecated_call(f)
 
 
 class TestWarns(object):
@@ -181,4 +185,3 @@ class TestWarns(object):
         ''')
         result = testdir.runpytest()
         result.stdout.fnmatch_lines(['*2 passed in*'])
-

